tyssue/core/sheet.py

```diff
@@ -273,6 +275,75 @@ def extract_bounding_box(
         subsheet.reset_topo()
         return subsheet
 
+    def get_force_inference(self, coords=None):
+        """Measure force based on Brodland method.
+
+        g_gamma_matrix*tension_vector = 0
+        Equation is homogenous and to avoid tension_vectors = 0,
+        Construction and solve the constrained least-squares equation system
+        [[g_gamma_matrix.T g_gamma_matrix, C^T_1],
+         [C_1, 0]]
+         où C1 = {1....1}
+
+        """
+        if coords is None:
+            coords = self.coords
+        ndim = self.dim
+        dcoords = ["d" + c for c in coords]
+
+        self.get_extra_indices()
+        edges_index = self.east_edges
+        edges_index_opposite = self.west_edges
+
+        n_edges = len(edges_index)
+
+        srce, trgt = self.edge_df.loc[
+            edges_index, ['srce', 'trgt']].iteritems()
+        srce = srce[1].to_numpy()
+        trgt = trgt[1].to_numpy()
+
+        # Fill gamma matrix to measure tension
+        pos = (self.edge_df.loc[edges_index, dcoords].to_numpy() / np.repeat(np.linalg.norm(
+            self.edge_df.loc[edges_index, dcoords].to_numpy(), axis=1), ndim).reshape((n_edges, ndim)))
```

Comment:
This is computed in self.edge_df[["ux", "uy", "uz"]]] but this is done [here](https://github.com/DamCB/tyssue/blob/master/tyssue/dynamics/factory.py#L112) - which is not really good. Can you use `ucoords` here and move the computation in `factory.py` to `base_geometry`?

Reply:
As I understand, I add a method `update_ucoords` in base geometry?

Reply:
yes please :)

Reply:
I spend so much time to this line, and finally I just have to use `self.edge_df[["ux", "uy", "uz"]]]`...
Pourquoi faire simple quand on peut faire compliqué... --'

Reply:
:smile: